File: src/converter/plugins/convertNamedTupleMember.ts
import ts from "typescript";
import {createSimplePlugin} from "../plugin.js";
import {CheckCoverageService, checkCoverageServiceKey} from "./CheckCoveragePlugin.js";

export const convertNamedTupleMember = createSimplePlugin((node, context, render) => {
    if (!ts.isNamedTupleMember(node)) return null

    const checkCoverageService = context.lookupService<CheckCoverageService>(checkCoverageServiceKey)

    checkCoverageService?.cover(node)

    const name = render(node.name)
    const type = render(node.type)

    return `/* ${name}: */ ${type}`
})

File: gradle-plugin/karakum/src/jsMain/kotlin/io/github/sgrishchenko/karakum/extension/plugins/convertNamedTupleMember.kt
package io.github.sgrishchenko.karakum.extension.plugins

import io.github.sgrishchenko.karakum.extension.createSimplePlugin
import typescript.Node
import typescript.isNamedTupleMember

val convertNamedTupleMember = createSimplePlugin plugin@{ node: Node, context, render ->
    if (!isNamedTupleMember(node)) return@plugin null

    val checkCoverageService = context.lookupService<CheckCoverageService>(checkCoverageServiceKey)

    checkCoverageService?.cover(node)

    val name = render(node.name)
    val type = render(node.type)

    "/* ${name}: */ $type"
}
